### DISCUSSION TITLE
Correct usage of mainEntityOfPage in Article type

### DISCUSSION THREAD

**Discussion by uPagge** [schemaorg/schemaorg#3274] [Q&A] [answered]
I write a blog using Ghost CMS they automatically generate json-id. I also use a stats system that tries to use that json-id to show detailed stats on material.

The problem is that one of them is not interpreting the schema.org specification correctly. Help me understand who is wrong.

The analytics developers say that the mainEntityOfPage should contain a link to the article itself in @id.

The Ghost CMS developers believe that @id in mainEntityOfPage should point to the root page of the blog, not to the current article.

Would be grateful for help

**Comment by RichardWallis** [comment, marked as answer]
Following the description of the property at [schema.org/mainEntityOfPage](https://schema.org/mainEntityOfPage), the associated [background documentation](https://schema.org/docs/datamodel.html#mainEntityBackground), and [json-ld example](https://schema.org/mainEntityOfPage#eg-0383), the _Ghost CMS developers_ are not correct.

The _mainEntityOfPage_ property of an Article/BlogPosting should contain the url of the page (preferred page if it appears on more than one) it is displayed upon.

Having said that, reading your description, I do not think that the analytics developers have it quite correct either.

In the example you describe there are two things (entities) the Article/BlogPosting and the WebPage it is displayed upon.  Ideally they should be described separately with the mainEntityOfPage property of the Article linking to the WebPage - see the examples for [BlogPosting](https://schema.org/BlogPosting#examples).

For completeness, in addition to reviewing the use of _mainEntityOfPage_ to link an Article to the WebPage it is displayed upon, use of  its inverse [mainEntity](https://schema.org/mainEntity) should be looked at.   _mainEntity_ is a property of a CreativeWork (usually WebPage) linking to the main Thing (entity) that page describes - see [example](https://schema.org/mainEntity#eg-0382).

**Reply by uPagge** [reply]
Thank you very much for your detailed answer
